src/librustc/ty/print/pretty.rs

```diff
@@ -931,6 +945,8 @@ pub struct FmtPrinterData<'a, 'gcx, 'tcx, F> {
     binder_depth: usize,
 
     pub region_highlight_mode: RegionHighlightMode,
+
+    pub name_resolver: Option<Box<&'a dyn Fn(ty::sty::TyVid) -> Option<String>>>,
```

Comment:
Please cc me on `ty::print` changes - but also, I have a few questions, such as "why is this a `Box<&dyn Trait>`?" - you can just remove the `Box` and everything would still work.